allennlp/training/callbacks/callback.py

```diff
@@ -0,0 +1,29 @@
+from typing import TypeVar, Generic
+
+from allennlp.common.registrable import Registrable
+
+TrainerType = TypeVar('TrainerType')  # pylint: disable=invalid-name
+
+class Callback(Registrable, Generic[TrainerType]):
+    # Lower priority comes first
+    priority = 0
```

Comment:
Having this be a class attribute is a little opaque for a user implementing their own subclass - why not just have it be a constructor argument with a default value?

Reply:
I agree with this.